src/server/bg_services/namespace_monitor.js

```diff
@@ -46,96 +54,116 @@ class NamespaceMonitor {
     async test_namespace_resources_validity() {
 
         await P.map_with_concurrency(10, system_store.data.namespace_resources, async nsr => {
+            if (!this.should_monitor(nsr)) return;
+
+            const endpoint_type = nsr.nsfs_config ? 'NSFS' : nsr.connection;
+
             try {
-                if (!this.nsr_connections_obj[nsr._id]) {
-                    this.init_nsr_connection_to_target(nsr);
+                if (endpoint_type === 'NSFS') {
+                    await this.test_nsfs_resource(nsr);
+                } else if (['AWS', 'S3_COMPATIBLE', 'IBM_COS'].includes(endpoint_type)) {
```

Comment:
In the case of non-NSFS, the connection holds the type inside the endpoint_type variable.
It is an object, not a type.
Just take the endpoint_type from the connection above when you assign.